test/helper/ConversationGenerator.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {faker} from '@faker-js/faker';
+import {ConnectionStatus} from '@wireapp/api-client/lib/connection/';
+import {CONVERSATION_TYPE, CONVERSATION_ACCESS_ROLE} from '@wireapp/api-client/lib/conversation/';
+import {RECEIPT_MODE} from '@wireapp/api-client/lib/conversation/data';
+import {ConversationProtocol} from '@wireapp/api-client/lib/conversation/NewConversation';
+import {QualifiedId} from '@wireapp/api-client/lib/user';
+import {LegalHoldStatus} from '@wireapp/core/lib/conversation/content';
+
+import {ConnectionEntity} from 'src/script/connection/ConnectionEntity';
+import {ConversationDatabaseData, ConversationMapper} from 'src/script/conversation/ConversationMapper';
+import {ConversationStatus} from 'src/script/conversation/ConversationStatus';
+import {ConversationVerificationState} from 'src/script/conversation/ConversationVerificationState';
+import {Conversation} from 'src/script/entity/Conversation';
+import {User} from 'src/script/entity/User';
+import {createUuid} from 'Util/uuid';
+
+export function generateAPIConversation(
+  id: QualifiedId = {id: createUuid(), domain: 'test.wire.link'},
+  conversationType = CONVERSATION_TYPE.REGULAR,
+  conversationProtocol = ConversationProtocol.PROTEUS,
+  overwites?: Partial<ConversationDatabaseData>,
```

Comment:
I think we should use an object for those params, this allow settings only a few of them without doing `generateAPIConversation(undefined, undefined, PROTEUS)` for example. WDYT?

Reply:
https://github.com/wireapp/wire-webapp/pull/15611/commits/f5e2de1a522ffd58294088def3d83f46618b5539